enzy_htp/structure/residue.py

```diff
@@ -174,7 +183,7 @@ def is_missing_chain(self) -> bool:
 
     def is_sequence_eq(self, other: Residue) -> bool:
         """Comparator that checks for sequence same-ness."""
-        return self.key() == other.key()
+        return self.name == other.name
```

Comment:
This should by key() instead of name since it is trying to determine if the position is the same in sequence instead of if residues are the same type.

Reply:
I see. The idea is to make Chian.is_same_sequence work. That way I think this method name is missleading and I suggest making a new method like is_same_residue_name

Reply:
changed back to key()